### PR TITLE
feat: add `is_lock` and `is_allow` methods to `RotationConstraints`

### PR DESCRIPTION
**Pull request by LiamGallagher737**

### CODE REVIEW

File: core/src/constraints.rs [hunk 1]
```diff
@@ -57,6 +57,18 @@ impl RotationConstraints {
...
         }
     }
+    
+    /// Returns true if all axes are locked
+    #[must_use]
+    pub fn is_lock(&self) -> bool {
+        !self.allow_x && !self.allow_y && !self.allow_z
+    }
+
+    /// Returns true if all axes are nallowed
```
Comment (jcornaz) [inline core/src/constraints.rs:67]:
```suggestion
    /// Returns true if all axes are allowed (not locked)
```

File: core/src/constraints.rs [hunk 2]
```diff
@@ -57,6 +57,18 @@ impl RotationConstraints {
             allow_z: true,
         }
     }
+    
+    /// Returns true if all axes are locked
+    #[must_use]
+    pub fn is_lock(&self) -> bool {
```
Comment (jcornaz) [inline core/src/constraints.rs:63]:
```suggestion
    pub fn is_locked(&self) -> bool {
```

**Review by jcornaz — APPROVED** [review]
Thanks.

For the record, would you mind share why you needed this?

**Comment by LiamGallagher737** [comment]
> For the record, would you mind share why you needed this?

When doing the following in update the entity wouldn't move
```rs
if input.handbrake {
    *rotation_constraints = RotationConstraints::lock();
} else {
    *rotation_constraints = RotationConstraints::allow();
}
```
But this would
```rs
if input.handbrake && rotation_constraints.allow_x && rotation_constraints.allow_y && rotation_constraints.allow_z {
    *rotation_constraints = RotationConstraints::lock();
} else if !input.handbrake && !rotation_constraints.allow_x && !rotation_constraints.allow_y && !rotation_constraints.allow_z {
    *rotation_constraints = RotationConstraints::lock();
}
```

**Pull request merged**